# MemberId

In [1]:
reset

Once deleted, variables cannot be recovered. Proceed (y/[n])? y


In [2]:
import pandas as pd
df_Mem_ori = pd.read_csv('./data/Member.csv')
df_Mem_ori.head()

/home/sopper/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2728: DtypeWarning: Columns (8) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


MemberId Birthday GenderTypeDef RegisterSourceTypeDef  \
0   1587633      NaN           NaN                   Web   
1   1587634      NaN           NaN                   Web   
2   1587658      NaN           NaN                   Web   
3   1587809      NaN           NaN                   Web   
4   1587830      NaN           NaN                   Web   

          RegisterDateTime MinOrderDate  OpenCardPresent  \
0  2016-08-05 19:45:59.670          NaN                0   
1  2016-10-27 21:25:22.907          NaN                0   
2  2015-08-11 19:11:09.030          NaN                0   
3  2015-08-11 19:53:32.397          NaN                0   
4  2017-06-03 17:33:03.987          NaN                0   

   LastBirthdayPresentYear IsInBlackList  
0                        0           NaN  
1                        0           NaN  
2                        0           NaN  
3                        0           NaN  
4                        0           NaN

In [3]:
df_Mem = df_Mem_ori.copy()

In [4]:
df_Mem.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 859146 entries, 0 to 859145
Data columns (total 9 columns):
MemberId                   859146 non-null int64
Birthday                   237275 non-null object
GenderTypeDef              237276 non-null object
RegisterSourceTypeDef      829143 non-null object
RegisterDateTime           859146 non-null object
MinOrderDate               562300 non-null object
OpenCardPresent            859146 non-null int64
LastBirthdayPresentYear    859146 non-null int64
IsInBlackList              1004 non-null object
dtypes: int64(3), object(6)
memory usage: 59.0+ MB


## 需要的欄位
* MemberId
* GenderTypeDef
* RegisterSourceTypeDef
* OpenCardPresent
* IsInBlackList

## 檢視各欄位內容

In [5]:
# GT: GenderType
df_Mem['GenderTypeDef'].value_counts()

Female    234540
Male        2736
Name: GenderTypeDef, dtype: int64

In [6]:
# RST: RegisterSourceTypeDef
df_Mem['RegisterSourceTypeDef'].value_counts()

Web           624375
AndroidApp    109704
iOSApp         95064
Name: RegisterSourceTypeDef, dtype: int64

In [7]:
# OCP: OpenCardPresent
df_Mem['OpenCardPresent'].value_counts()

0        624888
1078      11997
1292      11092
1838      10618
630       10577
2664      10531
8596      10311
544        9982
10342      9443
2577       9278
467        9040
1438       8551
2294       8378
2051       7922
5489       7771
4974       7702
8052       6955
1626       6950
7693       6862
5493       6103
10788      5812
4559       5701
812        5065
383        4986
9175       4865
3581       4729
5938       4729
7282       4265
961        4199
6547       3921
6879       3314
190        3267
4183       3262
318        3042
11428      1414
173        1168
127         370
98           86
Name: OpenCardPresent, dtype: int64

In [8]:
# Black: IsInBlackList
df_Mem['IsInBlackList'].value_counts()

是    680
否    324
Name: IsInBlackList, dtype: int64

## 選定欄位之內容
* **MemberId**
    * NAN: 0
> 無須處理

* **GenderTypeDef**
    * NAN: 621870
    * Female: 234540
    * Male: 2736
> 新增兩個欄位 Female, Male

* **RegisterSourceTypeDef**
    * NAN: 3
    * Web: 624375
    * AndroidApp: 109704
    * iOSApp: 95064
> 更改內容 web->1, App->0

* **OpenCardPresent**
    * NAN: 0
    * 0: 624888
    * others: 234258
> 更改內容 0->1, others&NAN->0    

* **IsInBlackList**
    * NAN: 858142
    * 是: 680
    * 否: 324
> 更改內容 是->1, 否&NAN->0

In [9]:
df_Mem.drop(["Birthday", "RegisterDateTime", "MinOrderDate", "LastBirthdayPresentYear"], axis=1, inplace=True)
df_Mem.head()

MemberId GenderTypeDef RegisterSourceTypeDef  OpenCardPresent IsInBlackList
0   1587633           NaN                   Web                0           NaN
1   1587634           NaN                   Web                0           NaN
2   1587658           NaN                   Web                0           NaN
3   1587809           NaN                   Web                0           NaN
4   1587830           NaN                   Web                0           NaN

## GenderTypeDef

In [10]:
# 新增 Female, Male 兩個欄位
df_Mem["Female"] = 0
df_Mem["Male"] = 0

# 欄位轉換
df_Mem["Female"][df_Mem["GenderTypeDef"] == "Female"] = 1
df_Mem["Male"][df_Mem["GenderTypeDef"] == "Male"] = 1
# 踢掉舊欄位
df_Mem.drop("GenderTypeDef", axis=1, inplace=True)

# 結果檢驗
print("Female: \n", df_Mem['Female'].value_counts())
print("Male: \n", df_Mem['Male'].value_counts())

Female: 
 0    624606
1    234540
Name: Female, dtype: int64
Male: 
 0    856410
1      2736
Name: Male, dtype: int64


/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  
/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  import sys


## RegisterSourceTypeDef

In [11]:
df_Mem["RegisterSourceTypeDef"][df_Mem["RegisterSourceTypeDef"] == "Web"] = 1
df_Mem["RegisterSourceTypeDef"][df_Mem["RegisterSourceTypeDef"] != 1] = 0

df_Mem["RegisterSourceTypeDef"].value_counts()

/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


1    624375
0    234771
Name: RegisterSourceTypeDef, dtype: int64

## OpenCardPresent

In [12]:
df_Mem["OpenCardPresent"][df_Mem["OpenCardPresent"] == 0] = 1
df_Mem["OpenCardPresent"][df_Mem["OpenCardPresent"] != 1] = 0

df_Mem["OpenCardPresent"].value_counts()

/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


1    624888
0    234258
Name: OpenCardPresent, dtype: int64

## IsInBlackList

In [13]:
df_Mem["IsInBlackList"][df_Mem["IsInBlackList"] == "是"] = 1
df_Mem["IsInBlackList"][df_Mem["IsInBlackList"] != 1] = 0

df_Mem["IsInBlackList"].value_counts()

/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


0    858466
1       680
Name: IsInBlackList, dtype: int64

In [14]:
df_Mem.head()

MemberId RegisterSourceTypeDef  OpenCardPresent IsInBlackList  Female  Male
0   1587633                     1                1             0       0     0
1   1587634                     1                1             0       0     0
2   1587658                     1                1             0       0     0
3   1587809                     1                1             0       0     0
4   1587830                     1                1             0       0     0

In [16]:
df_Mem.to_csv("./data/proData/df_Mem.csv", index=False)

# Order

In [14]:
reset

Once deleted, variables cannot be recovered. Proceed (y/[n])? y


In [15]:
import pandas as pd
# O: Order
df_O = pd.read_csv('./data/Orders.csv')
df_O.head()

/home/sopper/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2728: DtypeWarning: Columns (23) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


DateId  MemberId                                     OrderGroupCode  \
0  20150703   1388133  b'\xd0_\x8eby\xe8\x88\xa5\x85\xe4\x15\nMC\xeaO...   
1  20150703   1388133  b'\xd0_\x8eby\xe8\x88\xa5\x85\xe4\x15\nMC\xeaO...   
2  20150703   1388133  b'\xd0_\x8eby\xe8\x88\xa5\x85\xe4\x15\nMC\xeaO...   
3  20150703   1388133  b'\xd0_\x8eby\xe8\x88\xa5\x85\xe4\x15\nMC\xeaO...   
4  20150703    365787  b"\x04?}pe$m\x7f\x85\xa0y\x81'e5\xca\xbb\xf9\x...   

  TrackSourceTypeDef TrackDeviceTypeDef PayProfileTypeDef  SalesOrderSlaveId  \
0                Web             Mobile       SevenEleven            4457777   
1                Web             Mobile       SevenEleven            4457778   
2                Web             Mobile       SevenEleven            4457779   
3                Web             Mobile       SevenEleven            4457780   
4                Web             Mobile       SevenEleven            4457791   

   SalePageId  IsMajor  IsGift   ...     SalesOrderReceiverId  City  District  \
0     1242871     True   False   ...                  1480890   嘉義縣       水上鄉   
1     1242880     True   False   ...                  1480890   嘉義縣       水上鄉   
2     1242880     True   False   ...                  1480890   嘉義縣       水上鄉   
3     1175574     True   False   ...                  1480890   嘉義縣       水上鄉   
4     1194488     True   False   ...                  1480896   桃園市       中壢區   

   ZipCode  StoreName  StoreId  DeliverTypeDef StatusDef  ReturnGoodsDateId  \
0    608.0       中庄門市   910828     SevenEleven    Finish                NaN   
1    608.0       中庄門市   910828     SevenEleven    Finish                NaN   
2    608.0       中庄門市   910828     SevenEleven    Finish                NaN   
3    608.0       中庄門市   910828     SevenEleven    Finish                NaN   
4    320.0       復華門市   114071     SevenEleven    Finish                NaN   

  CauseDef  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN  

[5 rows x 28 columns]

In [16]:
df_O.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5495276 entries, 0 to 5495275
Data columns (total 28 columns):
DateId                         int64
MemberId                       int64
OrderGroupCode                 object
TrackSourceTypeDef             object
TrackDeviceTypeDef             object
PayProfileTypeDef              object
SalesOrderSlaveId              int64
SalePageId                     int64
IsMajor                        bool
IsGift                         bool
IsSalePageGift                 bool
Quantity                       int64
UnitPrice                      float64
PromotionDiscount              float64
ECouponId                      int64
ECouponDiscount                float64
SalesOrderSlaveTotalPayment    float64
SalesOrderSlaveDateTime        object
SalesOrderReceiverId           int64
City                           object
District                       object
ZipCode                        float64
StoreName                      object
StoreId              

## 聚合運算
### 聚合方式
* by
    * MemberId
    * OrderGroupCode
* Item
    * DateId: first
    * TrackSourceTypeDef: first
    * TrackDeviceTypeDef: first
    * PromotionDiscount: sum
    * ECouponId: first
    * ECouponDiscount: sum
    * SalesOrderSlaveTotalPayment: sum
    * SalesOrderSlaveDateTime: first
    * City: first
    * StatusDef: first

In [17]:
# 將不需要的欄位拿掉
df_O.drop(["DateId", "PayProfileTypeDef", "SalesOrderSlaveId", "SalePageId", \
            "IsMajor", "IsGift", "IsSalePageGift", "Quantity", "UnitPrice", \
            "District", "ZipCode", "StoreName", "StoreId", "DeliverTypeDef", \
            "ReturnGoodsDateId", "CauseDef", "SalesOrderReceiverId"], axis=1, inplace=True)

In [18]:
# 定義各欄位需要的聚合方式
groupby_dict = {
    "TrackSourceTypeDef": "first",
    "TrackDeviceTypeDef": "first",
    "PromotionDiscount": "sum",
    "ECouponId": "first",
    "ECouponDiscount": "sum",
    "SalesOrderSlaveTotalPayment": "sum",
    "SalesOrderSlaveDateTime": "first",
    "City": "first",
    "StatusDef": "first"
}
data = df_O.groupby(["MemberId", "OrderGroupCode"]).aggregate(groupby_dict)

In [19]:
df_O_result = pd.DataFrame(data).reset_index()
df_O_result.head()

MemberId                                     OrderGroupCode  \
0      1326  b's\xc2 \xc2\x9bS\x91\xfe\x00l[\x95\x18\xca\x8...   
1      1329  b'\x1b\xe9E\xdc\x8a0\xf5\xce\xaa\xf6\x99L\xa1?...   
2      1329  b'\x9a\xdc\xaa\x1c\xedC1\xd5\x83k(\xfd\xc8\x9d...   
3      1329  b'f\x02\xc2\xbbB\xd7}\\uXe\x14D4\xa2\x07>O\xfa...   
4      1334  b'(\x87V\xea\x92Z\xc1\xca\xbd\x1b-\xb6\x80\xc3...   

  TrackSourceTypeDef TrackDeviceTypeDef  PromotionDiscount  ECouponId  \
0                Web                 PC                0.0          0   
1                Web                 PC                0.0          0   
2                Web                 PC                0.0          0   
3                Web                 PC                0.0          0   
4                Web             Mobile                0.0          0   

   ECouponDiscount  SalesOrderSlaveTotalPayment  SalesOrderSlaveDateTime City  \
0              0.0                        249.0  2016-07-15 12:04:25.547  台北市   
1              0.0                        420.0  2015-08-26 14:24:09.020  台北市   
2              0.0                        299.0  2015-08-31 18:32:54.320  台北市   
3              0.0                        199.0  2015-09-03 18:07:48.750  台北市   
4              0.0                        280.0  2015-11-18 18:42:51.193  台北市   

  StatusDef  
0    Cancel  
1    Cancel  
2    Cancel  
3    Cancel  
4    Cancel

## 檢視各欄位內容

In [20]:
check_list = ["TrackSourceTypeDef", "TrackDeviceTypeDef", "City", "StatusDef"]
for i in df_O_result:
    if i in check_list:
        print(i, "\n")
        print(df_O_result[i].value_counts())
        print("NAN:", df_O_result[i].isnull().sum())
        print("----------------------------------------")

TrackSourceTypeDef 

Web           817281
AndroidApp    272290
iOSApp        254363
Name: TrackSourceTypeDef, dtype: int64
NAN: 0
----------------------------------------
TrackDeviceTypeDef 

Mobile    1140763
PC         203097
Pad            74
Name: TrackDeviceTypeDef, dtype: int64
NAN: 0
----------------------------------------
City 

新北市    280053
台中市    194912
台北市    162884
桃園市    143239
高雄市    142018
台南市    100734
彰化縣     49809
新竹市     34922
屏東縣     32269
新竹縣     30720
苗栗縣     26422
雲林縣     22382
基隆市     21302
宜蘭縣     21296
南投縣     18975
嘉義縣     17372
花蓮縣     17180
嘉義市     13838
台東縣      8797
澎湖縣      4397
屏東市       201
台東市       122
桃園縣        21
1st        16
New        10
<fo         7
Nan         5
Tao         5
Tai         3
25          3
Kot         2
No.         2
Daf         2
Jia         2
Hsi         1
Ins         1
Keb         1
Hy          1
Quậ         1
Cha         1
Aya         1
Kao         1
Đại         1
Xin         1
Name: City, dtype: int64
NAN: 2
------------

## 選定欄位之內容
* **TrackSourceTypeDef**
    * NAN: 0
    * Web: 817281
    * AndroidApp: 272290
    * iOSApp: 254363
> 更改內容： Web->0, AndroidApp&iOSApp->1

* **TrackDeviceTypeDef**
    * Mobile: 1140763
    * PC: 203097
    * Pad: 74
> 更改內容： Mobile->0, PC&Pad->1

* **City**
    * 新北市: 280053
    * 台中市: 194912
    * 台北市: 162884
    * 桃園市: 143239
    * 高雄市: 142018
    * 台南市: 100734
    * 彰化縣:  49809
    * 新竹市:  34922
    * 屏東縣:  32269
    * 新竹縣:  30720
    * 苗栗縣:  26422
    * 雲林縣:  22382
    * 基隆市:  21302
    * 宜蘭縣:  21296
    * 南投縣:  18975
    * 嘉義縣:  17372
    * 花蓮縣:  17180
    * 嘉義市:  13838
    * 台東縣:   8797
    * 澎湖縣:   4397
    * 屏東市:    201
    * 台東市:    122
    * 桃園縣:     21
    * 1st: 16
    * New: 10
    * <fo: 7
    * Tao: 5
    * Nan: 5
    * Tai: 3
    * 25:  3
    * No.: 2
    * Daf: 2
    * Jia: 2
    * Kot: 2
    * Aya: 1
    * Keb: 1
    * Cha: 1
    * Kao: 1
    * Đại: 1
    * Ins: 1
    * Hy:  1
    * Hsi: 1
    * Quậ: 1
    * Xin: 1
    * NAN: 2
> 刪除缺失資料 
> 新增欄位：
        * 台北市: A
        * 台中市: B
        * 基隆市: C
        * 台南市: D
        * 高雄市: E
        * 新北市: F
        * 宜蘭縣: G
        * 桃園市: H
        * 嘉義市: I
        * 新竹縣: J
        * 苗栗縣: K
        * 台中市: L
        * 南投縣: M
        * 彰化縣: N
        * 新竹市: O
        * 雲林縣: P
        * 嘉義縣: Q
        * 台南縣: R
        * 高雄縣: S
        * 屏東縣: T
        * 花蓮縣: U
        * 台東縣: V
        * 金門縣: W
        * 澎湖縣: X
        * 陽明山: Y
        * 連江縣: Z

* **StatusDef**
    * Finish: 1018638
    * Cancel: 181725
    * Overdue: 81454
    * ReturnGoods: 61766
    * WaitingToShipping: 200
    * CreditCheckFail: 151
    * NAN: 0
> 更改內容： Finish->1, others->0

## TrackSourceTypeDef

In [21]:
# Web->0, AndroidApp&iOSApp->1
df_O_result["TrackSourceTypeDef"][df_O_result["TrackSourceTypeDef"] == "Web"] = 0
df_O_result["TrackSourceTypeDef"][df_O_result["TrackSourceTypeDef"] != 0] = 1

df_O_result["TrackSourceTypeDef"].value_counts()

/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  
/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


0    817281
1    526653
Name: TrackSourceTypeDef, dtype: int64

## TrackDeviceTypeDef

In [22]:
# Mobile->0, PC&Pad->1
df_O_result["TrackDeviceTypeDef"][df_O_result["TrackDeviceTypeDef"] == "Mobile"] = 0
df_O_result["TrackDeviceTypeDef"][df_O_result["TrackDeviceTypeDef"] != 0] = 1

df_O_result["TrackDeviceTypeDef"].value_counts()

/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  
/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


0    1140763
1     203171
Name: TrackDeviceTypeDef, dtype: int64

## City

In [23]:
# 刪除闕漏資料
country_dict = {
    "台北市": "A", 
    "台中市": "B", 
    "基隆市": "C", 
    "台南市": "D", 
    "高雄市": "E", 
    "新北市": "F", 
    "宜蘭縣": "G", 
    "桃園市": "H", 
    "嘉義市": "I", 
    "新竹縣": "J", 
    "苗栗縣": "K",
    "屏東市": "L",
    "南投縣": "M", 
    "彰化縣": "N", 
    "新竹市": "O", 
    "雲林縣": "P", 
    "嘉義縣": "Q", 
    "台南縣": "R", 
    "高雄縣": "S", 
    "屏東縣": "T", 
    "花蓮縣": "U", 
    "台東縣": "V", 
    "金門縣": "W", 
    "澎湖縣": "X", 
    "陽明山": "Y", 
    "連江縣": "Z", 
    "台東市": "AA",
    "桃園縣": "AB"
}

# for i in df_O_result['City']:
# #     print(i)
#     if i in country_dict.keys():
#         print("yes")
    
for i in country_dict:
    df_O_result["City"][df_O_result["City"] == i] = country_dict[i]

df_O_result["City"].value_counts()
# if "台市" in country_dict.keys():
#     print("yes")

/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:39: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


F      280053
B      194912
A      162884
H      143239
E      142018
D      100734
N       49809
O       34922
T       32269
J       30720
K       26422
P       22382
C       21302
G       21296
M       18975
Q       17372
U       17180
I       13838
V        8797
X        4397
L         201
AA        122
AB         21
1st        16
New        10
<fo         7
Tao         5
Nan         5
25          3
Tai         3
No.         2
Kot         2
Daf         2
Jia         2
Quậ         1
Aya         1
Đại         1
Kao         1
Keb         1
Cha         1
Hy          1
Hsi         1
Ins         1
Xin         1
Name: City, dtype: int64

In [24]:
# 建立新欄位 & 欄位轉換
for i in country_dict.values():
    df_O_result[i] = 0
    df_O_result[i][df_O_result["City"] == i] = 1
    
df_O_result.drop("City", axis=1, inplace=True)

/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  after removing the cwd from sys.path.


In [25]:
for i in country_dict.values():
    print(i, "\n", df_O_result[i].value_counts())
    print("---------------")

A 
 0    1181050
1     162884
Name: A, dtype: int64
---------------
B 
 0    1149022
1     194912
Name: B, dtype: int64
---------------
C 
 0    1322632
1      21302
Name: C, dtype: int64
---------------
D 
 0    1243200
1     100734
Name: D, dtype: int64
---------------
E 
 0    1201916
1     142018
Name: E, dtype: int64
---------------
F 
 0    1063881
1     280053
Name: F, dtype: int64
---------------
G 
 0    1322638
1      21296
Name: G, dtype: int64
---------------
H 
 0    1200695
1     143239
Name: H, dtype: int64
---------------
I 
 0    1330096
1      13838
Name: I, dtype: int64
---------------
J 
 0    1313214
1      30720
Name: J, dtype: int64
---------------
K 
 0    1317512
1      26422
Name: K, dtype: int64
---------------
L 
 0    1343733
1        201
Name: L, dtype: int64
---------------
M 
 0    1324959
1      18975
Name: M, dtype: int64
---------------
N 
 0    1294125
1      49809
Name: N, dtype: int64
---------------
O 
 0    1309012
1      34922
Name: O, dtype: in

## StatusDef

In [26]:
# Finish->1, others->0
df_O_result["StatusDef"][df_O_result["StatusDef"] == "Finish"] = 1
df_O_result["StatusDef"][df_O_result["StatusDef"] != 1] = 0

df_O_result["StatusDef"].value_counts()

/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  
/home/sopper/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


1    1018638
0     325296
Name: StatusDef, dtype: int64

In [27]:
df_O_result.to_csv("./data/proData/df_O.csv", index=False)